### PR TITLE
feat: add logic to cancel submit approvals buttons

### LINKED ISSUE #22748: Confirmation redesign - Personal sign, implement cancel / sign buttons
Implement logic for cancel / submit buttons

### PR DESCRIPTION
**Pull request by jpuri**
## **Description**

Implement cancel / Submit logic on personal sign page.

## **Related issues**

Fixes: MetaMask/metamask-extension#22748

## **Manual testing steps**

1. Enable confirmation redesign locally
2. Go to Test DAPP click personal sign button
3. Cancel / Submit the page - it should work as expected

## **Screenshots/Recordings**
https://github.com/MetaMask/metamask-extension/assets/2182307/0f846c05-4a1d-4056-b88d-3a1744f94f04

## **Pre-merge author checklist**

- [X] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [X] I've clearly explained what problem this PR is solving and how it is solved.
- [X] I've linked related issues
- [X] I've included manual testing steps
- [X] I've included screenshots/recordings if applicable
- [X] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [X] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.
- [X] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
  - [X] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Comment by eriknson** [comment]
So excited to see this! Noted a few things / nits in the design review below:

redesigned-signature-review

**Review by eriknson — CHANGES_REQUESTED** [review]
So excited to see this! Here's design review / QA:
redesigned-signature-review

**Comment by jpuri** [comment]
Hey @eriknson : update on the UI feedbacks:

1. Scroll down , account into modal are yet to be developer
2. Current origin is `https://` thus its is not showing `http` highlighted
3. Ethereum Symbol - I checked and we are using same svg icon as network select
4. Footer - I fixed this 

Screenshot 2024-02-13 at 5 12 27 PM

**Review by segun — DISMISSED** [review]

**Review by digiwand — DISMISSED** [review]

**Comment by jpuri** [comment]
@eriknson : This PR is adding logic to footer buttons not adding any UI element. Can you plz remove change request.

**Review by eriknson — DISMISSED** [review]
Postponing design review to a PR closer to production as per request by @jpuri

File: ui/pages/confirmations/components/confirm/footer/footer.tsx [hunk 1]
```diff
@@ -1,30 +1,55 @@
...
+} from '../../../../../store/actions';
 import { useI18nContext } from '../../../../../hooks/useI18nContext';
 
 const Footer = () => {
   const t = useI18nContext();
+  const currentConfirmation = useSelector(currentConfirmationSelector);
+  const dispatch = useDispatch();
+
+  const onCancel = useCallback(() => {
+    if (!currentConfirmation) {
```
Comment (pedronfigueiredo) [inline ui/pages/confirmations/components/confirm/footer/footer.tsx:24]: @jpuri How can `currentConfirmation` be falsy here? If the state can indeed be that, shouldn't we disable the buttons instead?

Comment (jpuri) [reply ui/pages/confirmations/components/confirm/footer/footer.tsx:24]: Hey @pedronfigueiredo : when the component is rendered initially `currentConfirmation` can be undefined for momentarily for just 1 render cycle (till the hook initialises current confirmation). The check above is to ensure that we do not get null error in that scenario.

**Review by pedronfigueiredo — APPROVED** [review]

**Review by blackdevelopa — APPROVED** [review]

**Pull request merged**